midend/interpreter.cpp

```diff
@@ -913,6 +932,114 @@ void ExpressionEvaluator::postorder(const IR::BoolLiteral* expression) {
     set(expression, new SymbolicBool(expression));
 }
 
+void ExpressionEvaluator::postorder(const IR::Lss* expression) {
+    auto l = get(expression->left);
+    auto clone = expression->clone();
+    if (l->is<SymbolicInteger>()) {
```

Comment:
I apologize, this statement was incorrect, since this check is about the type of the left operand, and not of the result.
However, this case should be already handled by Operation_Relation. There is a bug in that function, though, where `expression->apply(cf)` should be replaced by `clone->apply(cf)` (in two places). That is the correct fix. Your code here leaves aside many important cases which are handled by that function and it also duplicates code needlessly.